Library/Homebrew/lock_file.rb

```diff
@@ -11,14 +11,15 @@ class LockFile
   class OpenFileChangedOnDisk < RuntimeError; end
   private_constant :OpenFileChangedOnDisk
 
+  sig { returns(Pathname) }
   attr_reader :path
 
   sig { params(type: Symbol, locked_path: Pathname).void }
   def initialize(type, locked_path)
     @locked_path = locked_path
     lock_name = locked_path.basename.to_s
-    @path = HOMEBREW_LOCKS/"#{lock_name}.#{type}.lock"
-    @lockfile = nil
+    @path = T.let(HOMEBREW_LOCKS/"#{lock_name}.#{type}.lock", Pathname)
+    @lockfile = T.let(nil, T.nilable(File))
```

Comment:
Why not
```suggestion
    @lockfile = T.let(nil, T.nilable(Pathname))
```
?

This will let you keep the `path.open` call below.